vars/checksPublishResults.groovy

```diff
@@ -0,0 +1,240 @@
+                'unHealthy': null,
+                'thresholds': [
+                    'fail': ['all': null,'low': null,'normal': null,'high': null],
+                    'unstable': ['all': null,'low': null,'normal': null,'high': null]
+                ]
+            ],
+            'archive': null
+        ]
+        final Map stepDefaults = ConfigurationLoader.defaultStepConfiguration(script, STEP_NAME)
+        final Map stepConfiguration = ConfigurationLoader.stepConfiguration(script, STEP_NAME)
+        prepare(parameters)
+        Map configuration = ConfigurationMerger.mergeDeepStructure(parameters, configurationKeys, stepConfiguration, configurationKeys, stepDefaults)
+
+        def doArchive = configuration.get('archive')
+        // JAVA
+        report('PmdPublisher', configuration.get('pmd'), doArchive)
+        report('DryPublisher', configuration.get('cpd'), doArchive)
+        report('FindBugsPublisher', configuration.get('findbugs'), doArchive)
+        report('CheckStylePublisher', configuration.get('checkstyle'), doArchive)
+        // JAVA SCRIPT
+        reportWarnings('JSLint', configuration.get('eslint'), doArchive)
+        // PYTHON
+        reportWarnings('PyLint', configuration.get('pylint'), doArchive)
+        // GENERAL
+        reportTasks(configuration.get('tasks'))
+        aggregate(configuration.get('aggregation'))
+    }
+}
+
+def aggregate(settings){
+    if (settings.active) {
+        def options = createCommonOptionsMap('AnalysisPublisher', settings)
+        // publish
+        step(options)
+    }
+}
+
+def reportTasks(settings){
+    if (settings.active) {
+        def options = createCommonOptionsMap('TasksPublisher', settings)
+        options.put('pattern', settings.get('pattern'))
+        options.put('high', settings.get('high'))
+        options.put('normal', settings.get('normal'))
+        options.put('low', settings.get('low'))
+        // publish
+        step(options)
+    }
+}
+
+def report(publisherName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap(publisherName, settings)
+        options.put('pattern', pattern)
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+def reportWarnings(parserName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap('WarningsPublisher', settings)
+        options.put('parserConfigurations', [[
+            parserName: parserName,
+            pattern: pattern
+        ]])
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+@NonCPS
+def isMap(object){
+    return object in Map
+}
+
+@NonCPS
+def toMap(parameter, activeByDefault = false){
```

Comment:
Do we really need that dualism(Map, boolean)? I mean it is possible in groovy. But the fact that is is possible does not mean that it really needs to be done. Having this dualism results in code that is less easy to understand IMO, without real benefit, despit something like "Look here, what I can do ..."
In case this really simplies something, please explain.